src/pages/CardRegisterPage.tsx

```diff
@@ -0,0 +1,97 @@
+import { useCallback, useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { CardForm, CreditCard, TopNavigation } from '@/components';
+import { Button } from '@/components/UI';
+import { styled } from '@/lib/stitches.config';
+import { getItem, setItem } from '@/storage/storage';
+import { CardKey, CardObj } from '@/types';
+export const initialCardState = {
+  [CardKey.CARD_NUMBERS]: {
+    val: {
+      1: '',
+      2: '',
+      3: '',
+      4: '',
+    },
+    isValid: false,
+  },
+  [CardKey.EXPIRE_DATE]: {
+    val: {
+      month: '',
+      year: '',
+    },
+    isValid: false,
+  },
+  [CardKey.CVC]: {
+    val: '',
+    isValid: false,
+  },
+  [CardKey.PASSWORD]: {
+    val: '',
+    isValid: false,
+  },
+  [CardKey.OWNER_NAME]: {
+    val: '',
+    isValid: false,
+  },
+} as const;
+
+export const CardRegisterPage = () => {
+  const navigate = useNavigate();
+  const [cardState, setCardState] = useState(initialCardState);
+  const cardDisplayInfo = useMemo(
+    () => ({
+      [CardKey.CARD_NUMBERS]: cardState[CardKey.CARD_NUMBERS].val,
+      [CardKey.OWNER_NAME]: cardState[CardKey.OWNER_NAME].val,
+      [CardKey.EXPIRE_DATE]: cardState[CardKey.EXPIRE_DATE].val,
+    }),
+    [cardState]
+  );
+
+  const handleCardState = useCallback((key: CardKey, state: CardObj) => {
+    setCardState((prev) => ({
+      ...prev,
+      [key]: { ...state },
+    }));
+  }, []);
+
+  const isAllValid = Object.values(cardState).every((input) => input.isValid);
+
+  const handleSubmit = () => {
+    const newCard = Object.entries(cardState).reduce((acc, cur) => {
+      return { ...acc, [cur[0]]: cur[1].val };
+    }, {});
```

Comment:
이렇게 하면 가독성이 좀 더 좋아질 것 같아요 :)

```suggestion
    const newCard = Object.entries(cardState).reduce((cardState, [cardKey, form]) => {
      return { ...cardState, cardKey: form.val };
    }, {});
```